src/main/java/duke/storage/SaveParser.java

```diff
@@ -0,0 +1,170 @@
+package duke.storage;
+
+import duke.exceptions.DukeReadSaveException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.util.Scanner;
+
+/**
+ * Parser class to create Task objects from string commands.
+ */
+public class SaveParser {
+    private static final DukeReadSaveException MISSING_ARG_EXCEPTION =
+            new DukeReadSaveException("missing argument in save");
```

Comment:
I like this, quite smart to reuse a cached exception 👍

Reply:
Agree! Definitely will try this trick in my code in the future 👍